data/data/gcp/cluster/main.tf

```diff
@@ -107,3 +108,20 @@ resource "google_compute_image" "cluster" {
 
   licenses = var.gcp_image_licenses
 }
+
+data "google_compute_image" "cluster_images" {
+  count = var.gcp_preexisting_image ? 0 : 1
+  name  = "${var.cluster_id}-rhcos-image"
+
+  depends_on = [google_compute_image.cluster]
+}
```

Comment:
Why create a data source here? I think you can just read image_id from the cluster image

Reply:
Yeah, but `google_compute_image` API is not exporting `image_id` attribute. Hence had to make use of data source to fetch the `image_id` value. [Link](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_image.html#attributes-reference) has details on the list of attributes that are made available by the API.